controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -439,6 +440,26 @@ private long findSegmentSplitsMerges(List<Segment> referenceSegmentsList, List<S
                                                         segmentIndex.getData(), segmentTable.getData(), from, to)))));
     }
 
+    @Override
+    public CompletableFuture<Boolean> isStreamCutValid(Map<Long, Long> streamCut) {
+        return Futures.allOfWithResults(streamCut.keySet().stream().map(x -> getSegment(x).thenApply(segment ->
+                new SimpleEntry<>(segment.getKeyStart(), segment.getKeyEnd())))
+                .collect(Collectors.toList()))
+                .thenAccept(TableHelper::validateStreamCut)
+                .handle((r, e) -> {
+                    if (e != null) {
+                        if (Exceptions.unwrap(e) instanceof IllegalArgumentException) {
+                            return false;
+                        } else {
+                            log.warn("Exception while trying to validate a stream cut for stream {}/{}", scope, name);
+                            throw Lombok.sneakyThrow(e);
```

Comment:
we could throw the unwrapped exception, right?

Reply:
That will be a checked exception and we would have to catch it in that case.